userservice/username.py

```diff
@@ -0,0 +1,24 @@
+import re
+from django.conf import settings
+
+
+def _get_format():
+    return getattr(settings, "REMOTE_USER_FORMAT", "eppn")
+
+
+def expect_uid():
+    username_format = _get_format()
+    return username_format is not None and\
+        (username_format == "uid" or username_format == "uwnetid")
+
+
+EPPN_PATTERN =\
+    re.compile(r'^([^@]+)@[a-z0-9_-]+\.[a-z]+$', re.I)
+
+
+def get_uid(username):
+    if username is not None and len(username) and expect_uid():
+        found = re.match(EPPN_PATTERN, username)
```

Comment:
I'd be tempted to use a split() here instead of a regex:
`try:`
`        (username, domain) = username.split('@', 1)`
`except ValueError as ex:`
         `pass`